File: 10828/10828.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   10828.cpp                                          :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2024/02/25 17:19:43 by jihoolee          #+#    #+#             */
/*   Updated: 2024/02/25 19:11:09 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <string>
#include <stack>

int pop_stack(std::stack<int>* s) {
  if (s->empty())
    return -1;
  int result = s->top();

  s->pop();
  return result;
}

void print_top(std::stack<int>* s) {
  if (s->empty())
    std::cout << -1 << '\n';
  else
    std::cout << s->top() << '\n';
}

void do_cmd(std::stack<int>* s, const std::string& cmd) {
  int num;

  if (cmd == "push") {
    std::cin >> num;
    s->push(num);
  } else if (cmd == "pop") {
    std::cout << pop_stack(s) << '\n';
  } else if (cmd == "size") {
    std::cout << s->size() << '\n';
  } else if (cmd == "empty") {
    std::cout << int(s->empty()) << '\n';
  } else if (cmd == "top") {
    print_top(s);
  }
}

int main(void) {
  std::ios::sync_with_stdio(false);
  std::cin.tie(nullptr);
  std::cout.tie(nullptr);

  int             cmd_cnt;
  std::string     cmd;
  std::stack<int> s;

  std::cin >> cmd_cnt;
  for (int i = 0; i < cmd_cnt; ++i) {
    std::cin >> cmd;
    do_cmd(&s, cmd);
  }
  return 0;
}

File: 10828/10828.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    10828.py                                           :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2024/02/25 19:11:42 by jihoolee          #+#    #+#              #
#    Updated: 2024/03/02 15:46:41 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

import sys


class Stack:
    def __init__(self):
        self.container: list[int] = []
        self.size: int = 0

    def push(self, num: int) -> None:
        self.container.append(num)
        self.size += 1

    def pop(self) -> int:
        if (self.empty()):
            return -1
        result: int = self.container.pop()
        self.size -= 1
        return result

    def empty(self) -> bool:
        if (self.size == 0):
            return True
        else:
            return False

    def top(self) -> int:
        if (self.empty()):
            return -1
        return self.container[-1]


def main() -> None:
    s: Stack = Stack()

    for _ in range(int(input())):
        cmd: list[str] = list(sys.stdin.readline().strip().split())

        if cmd[0] == 'push':
            s.push(int(cmd[1]))
        elif cmd[0] == 'pop':
            print(s.pop())
        elif cmd[0] == 'size':
            print(s.size)
        elif cmd[0] == 'empty':
            print(int(s.empty()))
        elif cmd[0] == 'top':
            print(s.top())
        else:
            raise ValueError('Command can must be number between 1-5')


if __name__ == '__main__':
    main()
